test/jasmine/tests/select_test.js

```diff
@@ -54,6 +56,80 @@ describe('select box and lasso', function() {
         expect(actual.y).toBeCloseToArray(expected.y, PRECISION);
     }
 
+    describe('select elements', function() {
+        var mockCopy = Lib.extendDeep({}, mock);
+        mockCopy.layout.dragmode = 'select';
+
+        var gd;
+        beforeEach(function(done) {
+            gd = createGraphDiv();
+
+            Plotly.plot(gd, mockCopy.data, mockCopy.layout)
+                .then(done);
+        });
+
+        it('should be appended to the zoom layer', function() {
+            var x0 = 100;
+            var y0 = 200;
+            var x1 = 150;
+            var y1 = 200;
+
+            mouseEvent('mousemove', x0, y0);
+            expect(d3.selectAll('.zoomlayer > .zoombox-corners').size())
+                .toEqual(0);
+
+            mouseEvent('mousedown', x0, y0);
+            mouseEvent('mousemove', x1, y1);
+            expect(d3.selectAll('.zoomlayer > .zoombox-corners').size())
+                .toEqual(1);
+            expect(d3.selectAll('.zoomlayer > .select-outline').size())
+                .toEqual(2);
+
+            mouseEvent('mouseup', x1, y1);
+            expect(d3.selectAll('.zoomlayer > .zoombox-corners').size())
+                .toEqual(0);
+            expect(d3.selectAll('.zoomlayer > .select-outline').size())
+                .toEqual(2);
```

Comment:
maybe add an assertion after `doubleClick` making sure that the `select-outline` are cleared.